episode-11/chunk.py

```diff
@@ -35,30 +34,37 @@ def __init__(self, world, chunk_position):
 
 		# mesh variables
 
-		self.mesh_vertex_positions = []
-		self.mesh_tex_coords = []
-		self.mesh_shading_values = []
+		self.mesh = []
+		self.translucent_mesh = []
 
-		self.mesh_index_counter = 0
-		self.mesh_indices = []
+		self.mesh_quad_count = 0
+		self.translucent_quad_count = 0
 
 		# create VAO and VBO's
 
 		self.vao = gl.GLuint(0)
-		gl.glGenVertexArrays(1, self.vao)
+		gl.glGenVertexArrays(1, ctypes.byref(self.vao))
 		gl.glBindVertexArray(self.vao)
 
-		self.vertex_position_vbo = gl.GLuint(0)
-		gl.glGenBuffers(1, self.vertex_position_vbo)
+		self.vbo = gl.GLuint(0)
+		gl.glGenBuffers(1, ctypes.byref(self.vbo))
+		gl.glBindBuffer(gl.GL_ARRAY_BUFFER, self.vbo)
 
-		self.tex_coord_vbo = gl.GLuint(0)
-		gl.glGenBuffers(1, self.tex_coord_vbo)
+		gl.glVertexAttribPointer(0, 3, gl.GL_FLOAT, 
+				gl.GL_FALSE, 7 * ctypes.sizeof(gl.GLfloat), 0)
+		gl.glEnableVertexAttribArray(0)
+		gl.glVertexAttribPointer(1, 3, gl.GL_FLOAT, 
+				gl.GL_FALSE, 7 * ctypes.sizeof(gl.GLfloat), 3 * ctypes.sizeof(gl.GLfloat))
+		gl.glEnableVertexAttribArray(1)
+		gl.glVertexAttribPointer(2, 1, gl.GL_FLOAT, 
+				gl.GL_FALSE, 7 * ctypes.sizeof(gl.GLfloat), 6 * ctypes.sizeof(gl.GLfloat))
+		gl.glEnableVertexAttribArray(2)
 
-		self.shading_values_vbo = gl.GLuint(0)
-		gl.glGenBuffers(1, self.shading_values_vbo)
+		gl.glBindBuffer(gl.GL_ELEMENT_ARRAY_BUFFER, world.ibo)
 
-		self.ibo = gl.GLuint(0)
-		gl.glGenBuffers(1, self.ibo)
+	def __del__(self):
```

Comment:
I should probably retroactively add this to all episodes for correctness. I didn't bother during the tutorials because it's not technically necessary in this case and it's perhaps slightly confusing to newcomers.